### PR TITLE
kvserver: add metrics for Entries bytes read

### PR DESCRIPTION
**Pull request by tbg**
I used versions of these for https://github.com/cockroachdb/cockroach/pull/98576, and they are generally useful.

- kvserver: add raft.storage.read_bytes
- raftentry: add raft.entrycache.read_bytes

Part of #97917.

Release note: None
Epic: None

### CODE REVIEW

File: pkg/kv/kvserver/metrics.go [hunk 1]
```diff
@@ -979,6 +979,34 @@ of processing.
 		Measurement: "Elections called after timeout",
 		Unit:        metric.Unit_COUNT,
 	}
+	metaRaftStorageReadBytes = metric.Metadata{
+		Name: "raft.storage.read_bytes",
+		Help: `Counter of raftpb.Entry.Size() read from pebble for raft log entries.
+
+These are the bytes returned from the (raft.Storage).Entries method that were not
+returned via the raft entry cache. This metric plus the raft.storage.read_bytes
```
Comment (pavelkalinnikov) [inline pkg/kv/kvserver/metrics.go:987]:
```suggestion
returned via the raft entry cache. This metric plus the raft.entrycache.read_bytes
```

File: pkg/kv/kvserver/logstore/logstore.go [hunk 2]
```diff
@@ -496,9 +496,9 @@ func LoadEntries(
 	eCache *raftentry.Cache,
 	sideloaded SideloadStorage,
 	lo, hi, maxBytes uint64,
-) ([]raftpb.Entry, error) {
+) (_ []raftpb.Entry, _cachedSize int, _loadedSize int, _ error) {
```
Comment (pavelkalinnikov) [inline pkg/kv/kvserver/logstore/logstore.go:499]: Return `uint64` or `int64` to avoid int downcasts below?

**Review by pav-kv — APPROVED** [review]

**Comment by pavelkalinnikov** [comment]
Tag #97917?

File: pkg/kv/kvserver/logstore/logstore.go (hunk 2 above)
Comment (tbg) [reply pkg/kv/kvserver/logstore/logstore.go]: Done.

**Comment by tbg** [comment]
blathers backport 23.1

**Pull request merged**